sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java

```diff
@@ -2175,8 +2175,13 @@ private void translateWriteHelper(
           // If user does not specify a project we assume the table to be located in the project
           // that owns the Dataflow job.
           String projectIdFromOptions = context.getPipelineOptions().getProject();
-          LOG.warn(String.format(BigQueryIO.SET_PROJECT_FROM_OPTIONS_WARNING, table.getDatasetId(),
-              table.getDatasetId(), table.getTableId(), projectIdFromOptions));
```

Comment:
in case it's not obvious after reformatting, the bug was in the double use of `table.getDatasetId()`.